code-challenges.test.js

```diff
@@ -14,45 +14,128 @@
+//             // Expected output: "G0bbl3dyg00k"
+//         const secretCodeWord3 = "Eccentric"
+//             // Expected output: "3cc3ntr1c"
+//         expect(codedMessage(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+//         expect(codedMessage(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+//         expect(codedMessage(secretCodeWord3)).toEqual("3cc3ntr1c")
+//     })
+// })
 
-// Pseudo code:
 
-// --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
 
-// a) Create a test with expects statement using the variable provided.
+// // b) Create the function that makes the test pass.
 
-const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+// // Pseudo code:
+ 
+// // 1.) Declare a function with a parameter(string)
 
-const filterLetterA = "a"
-// Expected output: ["Mango", "Apricot", "Peach"]
-const filterLetterE = "e"
-// Expected output: ["Cherry", "Blueberry", "Peach"]
+// const codedMessage = (string) => {
+
+// // 2.) After creating the function, we need to find a built-in method that can change the requested values in any string we add to this function.
+
+// return string.replaceAll("a", "4").replaceAll("e", "3").replaceAll("i", "1").replaceAll("o", "0").replaceAll("E", "3");
+
+// // I would put more time into not hard coding the capitol "E" and to have a dynamic code to where if I put a character in it'll change whether capitolized or not.
+    
+    
+// }
+
+// // codedMessage(secretCodeWord1)
+// // codedMessage(secretCodeWord2)
+// // codedMessage(secretCodeWord3)
+
+// // --------------------2) Create a function that takes in an array of words and a single letter and returns an array of all the words containing that particular letter.
+
+// // a) Create a test with expects statement using the variable provided.
+
+// describe("allWords", () => {
+//     it("returns an array of all the words containing that particular letter.", () => {
+//         expect(allWords(fruitArray, filterLetterA)).toEqual(["Mango", "Apricot", "Peach"])
+//         expect(allWords(fruitArray, filterLetterE)).toEqual(["Cherry", "Blueberry", "Peach"])
+//     })
+// })
+
+// const fruitArray = ["Mango", "Cherry", "Apricot", "Blueberry", "Peach", "Kiwi"]
+
+// const filterLetterA = "a"
+// // Expected output: ["Mango", "Apricot", "Peach"]
+// const filterLetterE = "e"
+// // Expected output: ["Cherry", "Blueberry", "Peach"]
+
+// // b) Create the function that makes the test pass.
+
+// // pseudocode
+
+// // 1.) Declare the allWords function to take in the array.
+
+// const allWords = (word, letter) => {
+
+// // 2.) Return the .filter() HOF that takes the value placeholder thats required. We must create logic that iterates over the elements in the array. The logic also has to identify the placeholder of the specified characters and filter out the element containing that character.
```

Comment:
I suggest saying something like this instead: Use .filter to iterate and return values that include the letter parameter